ports/stm32/boards/PYBD_SF6/mpconfigboard.h

```diff
@@ -69,3 +71,50 @@
 #define MICROPY_HW_ETH_RMII_TX_EN   (pyb_pin_W8)
 #define MICROPY_HW_ETH_RMII_TXD0    (pyb_pin_W45)
 #define MICROPY_HW_ETH_RMII_TXD1    (pyb_pin_W49)
+
+// Reconfigure SPI flash for dynamic size detection.
+
+#undef MICROPY_BOARD_EARLY_INIT
+#undef MICROPY_HW_SPIFLASH_SIZE_BITS
+#undef MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2
+#undef MBOOT_SPIFLASH_BYTE_SIZE
+#undef MBOOT_SPIFLASH_LAYOUT
+#undef MBOOT_SPIFLASH_ERASE_BLOCKS_PER_PAGE
+#undef MBOOT_SPIFLASH2_BYTE_SIZE
+#undef MBOOT_SPIFLASH2_LAYOUT
+#undef MBOOT_SPIFLASH2_ERASE_BLOCKS_PER_PAGE
+
+#define MICROPY_BOARD_EARLY_INIT    board_early_init_sf6
+
+#define MICROPY_HW_SPIFLASH_DETECT_DEVICE (1)
+#define MICROPY_HW_SPIFLASH_QREAD_NUM_DUMMY(spiflash) (chip_params[spiflash->config == &spiflash_config ? 0 : 1]->qread_num_dummy)
```

Comment:
Having to look up entries in the 2-element `chip_params` array seems a little awkward.

Did you look at allowing a board to optionally enable a `const mp_spiflash_chip_params_t *params` member in `mp_spiflash_config_t`? That would allow all these ternary expressions to become `spiflash->config->params`, I think?)

Reply:
It can't be in `mp_spiflash_config_t` because that's a `const` struct that defines the *bus* and not the thing attached to the bus.  That struct would need to become non-const, and its definition included in `mpconfigboard.h`.

Reply:
I did try putting such an element in the `mp_spiflash_t` struct, but it means that `drivers/memory/spiflash.h` needs to be included in `mpconfigboard.h` which gets messy (but it's doable).

Reply:
Oh, my mistake.

What about having a board-level flag like `define MICROPY_HW_SPIFLASH_PARAMS 1` (or `define MICROPY_HW_SPIFLASH_PARAMS mp_spiflash_chip_params_t` if the structure definition is chosen by the board)?

That doesn't seem to create more inversion in the headers than this approach, does it?

Reply:
> What about having a board-level flag like `define MICROPY_HW_SPIFLASH_PARAMS 1`

Do you mean that that config option would add a `mp_spiflash_chip_params_t *` entry to `mp_spiflash_t`?

Reply:
> Do you mean that that config option would add a `mp_spiflash_chip_params_t *` entry to `mp_spiflash_t`?

Yes, like that.

Reply:
OK, this turned out quite good, and allowed me to move `mp_spiflash_chip_params_t` to `drivers/memory/spiflash.h`.

The trick to getting it working was to `#include "storage.h"` in `qspi.c`, so the latter knows about all the needed struct's (from `spi_bdev2`).  Because it's impossible to include `storage.h` in `mpconfigboard.h`, which is what tripped me up in my previous attempt.

Reply:
This approach looks good to me!